src/ripple/app/misc/impl/TxQ.cpp

```diff
@@ -1337,9 +1521,8 @@ TxQ::accept(Application& app,
                         ". Removing last item of account " <<
                         account.account;
                     auto endIter = byFee_.iterator_to(dropRIter->second);
-                    assert(endIter != candidateIter);
-                    erase(endIter);
-
+                    if (endIter != candidateIter)
+                        erase(endIter);
```

Comment:
I just looked at this block again, and I'm thinking about the reasons why dropping the last transaction was so important: When all you have is sequence transactions, you can't drop the current one, because you're guaranteed to wedge the later ones. If you drop the last sequence transaction, there's a chance, however small, for the others to succeed. Now with tickets, if the current transaction is ticket based, there's no reason not to drop the current ticket based transaction - it won't affect the ones "after" it.

So here's my suggestion. I'm not sure if it breaks any tests.

```
                if (account.dropPenalty && account.transactions.size() > 1 &&
                    isFull<95>())
                {
                    /* The queue is close to full, this account has multiple
                        txs queued, and this account has had a transaction
                        fail.
                    */
                    if (candidateIter->seqProxy.isTicket())
                        // Since this is a ticket, order does matter.
                        // Drop this one
                        candidateIter = eraseAndAdvance(candidateIter);
                    else
                    {
                        /* Even though we're giving this transaction another
                           chance, chances are it won't recover. So we don't
                           make things worse, drop the _last_ transaction for
                           this account.
                        */
                        auto dropRIter = account.transactions.rbegin();
                        assert(dropRIter->second.account == candidateIter->account);
                        JLOG(j_.warn()) << "Queue is nearly full, and transaction "
                                        << candidateIter->txID << " failed with "
                                        << transToken(txnResult)
                                        << ". Removing last item of account "
                                        << account.account;
                        auto endIter = byFee_.iterator_to(dropRIter->second);
                        if (endIter != candidateIter)
                            erase(endIter);
                        ++candidateIter;
                    }
                }
                else
                    ++candidateIter;
```